dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EnrollmentTrackerConverterService.java

```diff
@@ -159,10 +159,10 @@ private Enrollment from(
 
     if (!Objects.equal(previousStatus, dbEnrollment.getStatus())) {
       if (dbEnrollment.isCompleted()) {
-        dbEnrollment.setEndDate(new Date());
+        dbEnrollment.setEndDate(now);
         dbEnrollment.setCompletedBy(preheat.getUsername());
       } else if (dbEnrollment.getStatus().equals(ProgramStatus.CANCELLED)) {
```

Comment:
`Objects.equal` is implemented in a very null-protective way

https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained

Reply:
Since its supported by the language we should prefer it over any library 😄 See https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals